pkg/cloudprovider/providers/openstack/metadata.go

```diff
@@ -141,12 +151,24 @@ func getMetadataFromMetadataService() (*Metadata, error) {
 // Metadata is fixed for the current host, so cache the value process-wide
 var metadataCache *Metadata
 
-func getMetadata() (*Metadata, error) {
+func getMetadata(order []string) (*Metadata, error) {
 	if metadataCache == nil {
-		md, err := getMetadataFromConfigDrive()
-		if err != nil {
-			md, err = getMetadataFromMetadataService()
+		var md *Metadata
+		var err error
+
+		for _, id := range order {
+			switch id {
+			case configDriveID:
+				md, err = getMetadataFromConfigDrive()
+			case metadataID:
+				md, err = getMetadataFromMetadataService()
```

Comment:
Can we add 'default' case? If the order isn't configDrive and metadataService, err will be nil and break, that's bad.

Reply:
Added a default option.